tiledb/sm/rest/rest_client.cc

```diff
@@ -239,10 +239,8 @@ RestClient::get_array_schema_from_rest(const URI& uri) {
       ensure_json_null_delimited_string(&returned_data), nullopt);
   return {
       Status::Ok(),
-      make_shared<ArraySchema>(
-          HERE(),
-          serialization::array_schema_deserialize(
-              serialization_type_, returned_data))};
+      serialization::array_schema_deserialize(
```

Comment:
The `serialization::array_schema_deserialize` likely needs a MemoryTracker argument added so we can specify different types. I could see adding a MemoryTracker::Type::LOAD_SCHEMA or similar. Or even perhaps just re-use an `ANONYMOUS` tracker for the context?